dockers/docker-fpm-frr/base_image_files/monit_bgp

```diff
@@ -7,17 +7,17 @@
 ##  staticd
 ##  bgpcfgd
 ###############################################################################
-check process zebra matching "/usr/lib/frr/zebra"
-    if does not exist for 5 times within 5 cycles then alert
+check program bgp|zebra with path "/usr/bin/process_checker bgp /usr/lib/frr/zebra"
+    if status != 0 for 5 times within 5 cycles then alert
 
-check process fpmsyncd matching "fpmsyncd"
-    if does not exist for 5 times within 5 cycles then alert
+check program bgp|fpmsyncd with path "/usr/bin/process_checker bgp fpmsyncd"
+    if status != 0 for 5 times within 5 cycles then alert
 
-check process bgpd matching "/usr/lib/frr/bgpd"
-    if does not exist for 5 times within 5 cycles then alert
+check program bgp|bgpd with path "/usr/bin/process_checker bgp /usr/lib/frr/bgpd"
+    if status != 0 for 5 times within 5 cycles then alert
 
-check process staticd matching "/usr/lib/frr/staticd"
-    if does not exist for 5 times within 5 cycles then alert
+check program bgp|staticd with path "/usr/bin/process_checker bgp /usr/lib/frr/staticd"
+    if status != 0 for 5 times within 5 cycles then alert
 
-check process bgpcfgd matching "python /usr/local/bin/bgpcfgd"
-    if does not exist for 5 times within 5 cycles then alert
+check program bgp|bgpcfgd with path "/usr/bin/process_checker bgp /usr/bin/python /usr/local/bin/bgpcfgd"
```

Comment:
This command `/usr/bin/python /usr/local/bin/bgpcfgd` has a space in it. Is this tested? I believe you may need to wrap it in quotes.

Reply:
Yes, I tested this pattern on DUT and it did work. Because the string `/usr/bin/python /usr/local/bin/bgpcfgd` will be as a single parameter passed to the script `process_checker`. In other words, the content of variable `process_cmdline` will be `/usr/bin/python /usr/local/bin/bgpcfgd`.

Reply:
I'm a bit surprised by this, but as long as it's tested and working as expected, it's OK.

Reply:
Yes, that's the reason why it worked. I just wanted to give the detailed reason and found you already got it.